src/js/starter_app.js

```diff
@@ -0,0 +1,144 @@
+(function(window){
+  window.extractData = function() {
+    var ret = $.Deferred(); 
+  
+    function onError() {
+      console.log('Loading error', arguments);
+      ret.reject();
+    }     
+
+    function onReady(smart)  {
+      if (smart.hasOwnProperty('patient')) { 
+        var patient = smart.patient;
+        var pt = patient.read();
+        var obv = smart.patient.api.fetchAll({
+                      type: 'Observation', 
+                      query: {
+                        code: {
+                          $or: ['http://loinc.org|8302-2', 'http://loinc.org|8462-4',
+                                'http://loinc.org|8480-6', 'http://loinc.org|2085-9',
+                                'http://loinc.org|2089-1']
+                              }
+                             }
+                    });
+        
+        $.when(pt, obv).fail(onError);
+
+        $.when(pt, obv).done(function(patient, obv) {
+          var byCodes = smart.byCodes(obv, 'code');
+          var gender = patient.gender;
+          var dob = new Date(patient.birthDate);     
+          var day = dob.getDate(); 
+          var monthIndex = dob.getMonth() + 1;
+          var year = dob.getFullYear();
+
+          var dobStr = monthIndex + '/' + day + '/' + year;
+          var fname = '';
+          var lname = '';
+
+          if(typeof patient.name[0] !== 'undefined') {
+            fname = patient.name[0].given.join(' ');
+            lname = patient.name[0].family.join(' ');
+          }
+
+          var height = byCodes('8302-2');
+          var systolicbp = byCodes('8480-6');
+          var diastolicbp = byCodes('8462-4');
+          var hdl = byCodes('2085-9');
+          var ldl = byCodes('2089-1');
+
+          var p = defaultPatient();          
+          p.birthday = dobStr;
+          p.gender = gender;
+          p.fname = fname;
+          p.lname = lname;
+          p.age = parseInt(calculateAge(dob));
+
```

Comment:
Make sure that `valueQuantity.value` is valid and `valueQuantity.unit` is valid before concatenating.  If the unit is not present, you'd see `undefined` displayed.